Source/WebKit/UIProcess/wpe/ScreenManagerWPE.cpp

```diff
@@ -58,6 +73,11 @@ ScreenManager::ScreenManager()
         manager->updatePrimaryDisplayID();
         manager->propertiesDidChange();
     }), this);
+    g_signal_connect(display, "notify::available-input-devices", G_CALLBACK(+[](WPEDisplay* d, GParamSpec*, ScreenManager*) {
+        auto availableInputDevices = toAvailableInputDevices(wpe_display_get_available_input_devices(d));
+        for (auto& pool : WebProcessPool::allProcessPools())
+            pool->sendToAllProcesses(Messages::WebProcess::UpdateAvailableInputDevices(availableInputDevices));
+    }), this);
```

Comment:
I don't think this belongs here either. We could monitor the devices from the process pool and then notify all its processes. It's not common to have many process pools, so it's ok to monitor the same property from all of them. That way we don't need a common place for the conversion helper, it can be a static function in WebProcessPoolGLib.cpp. We can connect to the signal in WebProcessPool::platformInitialize

Reply:
And we should probably disconnect the signal too, in WebProcessPool::platformInvalidateContext probably